### ISSUE TITLE
Server dont update

### ISSUE THREAD

**Issue by arvan3844** [mornedhels/enshrouded-server#119]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Question

Hello,

i'm having problems with the update of the server. we got a new patch to a new build today, but the server dont update itself.
server logs says all ok:
`2024-12-11 12:15:52.739 supervisord: enshrouded-updater enshrouded-server: started`
`2024-12-11 12:15:52.768 supervisord: enshrouded-updater INFO - enshrouded-updater complete`

Server files are outdated (dated 07.12.2024)

Can't also use any of the commands, like
`docker compose exec enshrouded supervisorctl start enshrouded-force-update`
error is "no configuration file provided: not found". Running server on proxmox in a container.

So i used `supervirsorctl` directly in the console from the container and it worked, but it does not update the server files.
log: `supervisord: enshrouded-updater Error! App '2278520' state is 0x6 after update job.`

steamcmd says this to error 0x6: 
`Reason | No connection to content servers`
`Reason 2 | Received 401 (Unauthorized) HTTP response for depot 11`

Solution was the same as last time: delete both files `appmanifest_2278520.acf` and `appmanifest_228980.acf` in `steamapps` folder.

`PUID` and `PGID` are both correcly to 4711.

**Comment by kabixi000** [comment]
As a temporary workaround, I was able to resolve the issue by removing the following manifest files inside the container and restarting the server.

```bash
rm /opt/enshrouded/server/steamapps/appmanifest_228980.acf
rm /opt/enshrouded/server/steamapps/appmanifest_2278520.acf
```

**Comment by cp-fabian-pittroff** [comment]
Hello there,

seems like there went something wrong with the last steam update job.

> Can't also use any of the commands, like
> `docker compose exec enshrouded supervisorctl start enshrouded-force-update`
> error is "no configuration file provided: not found". Running server on proxmox in a container.

`docker compose exec ...` can only work if you are in the same directory as the `docker-compose.yaml`. The error indicates, that it can't find the file. You can also run `docker exec <name_of_container|default enshrouded> supervisor ...`.

> So i used `supervirsorctl` directly in the console from the container and it worked, but it does not update the server files. log: `supervisord: enshrouded-updater Error! App '2278520' state is 0x6 after update job.`
> 
> steamcmd says this to error 0x6: `Reason | No connection to content servers` `Reason 2 | Received 401 (Unauthorized) HTTP response for depot 11`
> 
> Solution was the same as last time: delete both files `appmanifest_2278520.acf` and `appmanifest_228980.acf` in `steamapps` folder.

Thats odd, don't know why it helps to delete the steam appmanifest, but I will include the cleanup of those files in the force update command.

**Comment by arvan3844** [comment]
Though atm the `enshrouded-updater `don't really update the client if a new build exist, so the server can't be found on the server browser ingame, because its outdated. Because it still says it's up2date. But i can't find any logs about why, it should try to update itself, since it only said `INFO - enshrouded-updater complete` but should have find the new build. In the past, this just works.

**Comment by andrewgdunn** [comment]
I was going to post an issue that I suspected to be networking but may be this as well. Just to communicate that networking is debugged:

```
root@0bc1628fd5f1:/usr/local/etc/enshrouded# ss -tupln
Netid                   State                    Recv-Q                   Send-Q                                     Local Address:Port                                        Peer Address:Port                   Process                   
udp                     UNCONN                   0                        0                                                0.0.0.0:15637                                            0.0.0.0:*                                                
udp                     UNCONN                   0                        0                                                0.0.0.0:48079                                            0.0.0.0:*                                                
tcp                     LISTEN                   0                        0                                              127.0.0.1:39273                                            0.0.0.0:*    
```

```
[dungeon@storr ~]$ podman ps | grep enshrouded
0bc1628fd5f1  docker.io/mornedhels/enshrouded-server:latest  /usr/local/etc/en...  31 minutes ago  Up 31 minutes        0.0.0.0:15637->15637/udp                                   enshrouded
```

My players have been unable to see the instance for several days (started last week) and I wasn't able to test a local connection until today. I'm unable to connect to it locally. I assume this is because of a client/server mismatch?

**Comment by andrewgdunn** [comment]
Can confirm that after doing what @kabixi000 recommended above the server becomes available:

```
[dungeon@storr ~]podman unshare rm enshrouded/server/steamapps/appmanifest_22*
[dungeon@storr ~]$ systemctl --user restart container-enshrouded.service
```

**Comment by cp-fabian-pittroff** [comment]
Could someone check if the following line can be found in the logs “Download of Enshrouded server failed - aborting update”?

The updater should not set a new version, if the update itself fails.

**Comment by cp-fabian-pittroff** [comment]
I got an enshrouded update today for both of my running instances. Both are still up and running and I can connect to them. So the updater script seems only to have a problem, when the steamcli errors out. The problem, I can't really reproduce that locally.

For now I will add the workaround into the force-update command. I'll have to test a little bit, but will provide an update soon™️

```
2024-12-11 14:00:15.305 supervisord: enshrouded-updater Success! App '2278520' fully installed.
2024-12-11 14:00:15.446 supervisord: enshrouded-updater DEBUG - [53487] - [setCurrentVersion]: 16705286
```

**Comment by cp-fabian-pittroff** [comment]
Well on my develop environment I got the same error. Not sure why though. But the steamworks common redis had a betakey flag in it. After removing and updating the betakey flag was gone.

<details><summary>old manifest (228980)</summary>
<p>

```json
{
        "appid"         "228980"
        "Universe"              "1"
        "name"          "appid_228980"
        "StateFlags"            "6"
        "installdir"            "Steamworks Shared"
        "LastUpdated"           "1723292042"
        "LastPlayed"            "0"
        "SizeOnDisk"            "0"
        "StagingSize"           "0"
        "buildid"               "14615842"
        "LastOwner"             "0"
        "UpdateResult"          "0"
        "BytesToDownload"               "0"
        "BytesDownloaded"               "0"
        "BytesToStage"          "0"
        "BytesStaged"           "0"
        "TargetBuildID"         "0"
        "AutoUpdateBehavior"            "0"
        "AllowOtherDownloadsWhileRunning"               "0"
        "ScheduledAutoUpdate"           "0"
        "FullValidateAfterNextUpdate"           "1"
        "InstalledDepots"
        {
        }
        "UserConfig"
        {
                "BetaKey"               "public"
        }
        "MountedConfig"
        {
                "BetaKey"               "public"
        }
}
```

</p>
</details> 

<details><summary>new manifest (228980)</summary>
<p>

```json
{
        "appid"         "228980"
        "Universe"              "1"
        "name"          "Steamworks Common Redistributables"
        "StateFlags"            "2"
        "installdir"            "Steamworks Shared"
        "LastUpdated"           "0"
        "LastPlayed"            "0"
        "SizeOnDisk"            "0"
        "StagingSize"           "0"
        "buildid"               "0"
        "LastOwner"             "0"
        "UpdateResult"          "0"
        "BytesToDownload"               "0"
        "BytesDownloaded"               "0"
        "BytesToStage"          "0"
        "BytesStaged"           "0"
        "TargetBuildID"         "0"
        "AutoUpdateBehavior"            "0"
        "AllowOtherDownloadsWhileRunning"               "0"
        "ScheduledAutoUpdate"           "0"
        "InstalledDepots"
        {
        }
        "UserConfig"
        {
        }
        "MountedConfig"
        {
        }
}
```

</p>
</details>

**Comment by cp-fabian-pittroff** [comment]
Released with version [1.4.1](https://github.com/mornedhels/enshrouded-server/releases/tag/1.4.1). Thanks for reporting and solving the issue 👍
